pkg/providers/docker/config/template-md.yaml

```
@@ -33,6 +33,11 @@ kind: MachineDeployment
 metadata:
   name: {{.workerNodeGroupName}}
   namespace: {{.eksaSystemNamespace}}
+{{- if .autoscalingConfig }}
+  annotations:
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: "{{ printf "%#v" .autoscalingConfig.MinCount }}"
```

I don't think you need to convert to string here
```suggestion
    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: {{ .autoscalingConfig.MinCount }}
```

---

I may be missing something, but from what I can tell keys/values of annotations [need to be strings](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/): 
```
"Note: The keys and the values in the map must be strings. In other words, you cannot use numeric, boolean, list or other types for either the keys or the values."
```

That's why the quotes are there. But maybe I've missed something?

---

I removed the printf's. Somehow I couldn't get it to work without them earlier, but now the tests pass without a problem. Not sure what the problem was.

---

Yea Kubernetes annotation needs string. But we build the spec from template: byte array -> string yaml. I think it should not matter if you convert to string in the template or not. In the end it is going to be string format.